src/api/app/views/webui2/shared/_sign_up.html.haml

```diff
@@ -6,14 +6,21 @@
 - else
   = form_tag({ controller: 'user', action: 'register', method: :post }, class: 'sign-up', autocomplete: 'off') do
     .form-group
-      = text_field_tag 'login', nil, placeholder: 'Username', autocomplete: 'off', class: 'form-control'
+      = label_tag 'login', 'Username:'
+      %abbr.text-danger{ title: 'required' } *
+      = text_field_tag 'login', nil, placeholder: 'Username', autocomplete: 'off', class: 'form-control', required: true
     .form-group
-      = text_field_tag 'email', nil, placeholder: 'Email address', autocomplete: 'off', class: 'form-control'
+      = label_tag 'email', 'Email:'
+      %abbr.text-danger{ title: 'required' } *
+      = text_field_tag 'email', nil, placeholder: 'Email address', autocomplete: 'off', class: 'form-control', type: 'email', required: true
     .form-group
-      = password_field_tag :password, nil, id: 'pwd', placeholder: 'Enter a password', autocomplete: 'off', class: 'form-control'
+      = label_tag 'password', 'Password:'
+      %abbr.text-danger{ title: 'required' } *
+      = password_field_tag :password, nil, id: 'pwd', placeholder: 'Enter a password', autocomplete: 'off', class: 'form-control', required: true
     .form-group
+      = label_tag 'password_confirmation', 'Password confirmation:'
+      %abbr.text-danger{ title: 'required' } *
       = password_field_tag(:password_confirmation, nil, id: 'pwd_confirmation', placeholder: 'Password confirmation', autocomplete: 'off',
-        class: 'form-control')
+        class: 'form-control', required: true)
     = hidden_field_tag 'register', 'true'
-    .text-center
-      = submit_tag 'Sign Up', class: 'btn btn-primary'
+    = submit_tag "#{submit_btn_text}", class: 'btn btn-primary'
```

Comment:
The String interpolation here is not needed, so it should be `= submit_tag submit_btn_text, class: 'btn btn-primary'`.

I also prefer with parentheses, but this is a nitpick.

Reply:
done